connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -216,4 +252,99 @@ public long calculateTableSize(TableInfo tableInfo, Optional<String> filter) {
           "Querying table size was interrupted on the client side", e);
     }
   }
+
+  public TableInfo materializeQueryToTable(String querySql, int viewExpirationTimeInHours) {
+    TableId tableId = createDestinationTable(Optional.empty(), Optional.empty());
+    return materializeTable(querySql, tableId, viewExpirationTimeInHours);
+  }
+
+  public TableInfo materializeViewToTable(
+      String querySql, TableId viewId, int viewExpirationTimeInHours) {
+    TableId tableId =
+        createDestinationTable(
```

Comment:
I don't quite understand why we need to convert viewId to tableId here, a comment would be helpful.